google_guest_agent/snapshot_listener.go

```diff
@@ -0,0 +1,252 @@
+	case <-timer:
+		logger.Errorf("timeout while running script at %s", path)
+		// Exit code of the timeout command when it times out
+		return 124, &scriptsTimeoutError{path}
+	case err := <-done:
+		if err != nil {
+			exiterr := err.(*exec.ExitError)
+			status := exiterr.Sys().(syscall.WaitStatus)
+			logger.Errorf("error while running script at %s, return with exit code %d", path, status.ExitStatus())
+			return status.ExitStatus(), nil
+		}
+	}
+	return 0, nil
+}
+
+func handleRequest(config snapshotConfig, path string, disks string) (pb.AgentErrorCode, int) {
+	ec, err := runScript(path, disks, config.Timeout)
+	if os.IsNotExist(err) {
+		logger.Infof("script not found at %s: ", path)
+		return pb.AgentErrorCode_SCRIPT_NOT_FOUND, 0
+	}
+
+	if _, ok := err.(*scriptsTimeoutError); ok {
+		logger.Infof("script at: %s timed out", path)
+		return pb.AgentErrorCode_SCRIPT_TIMED_OUT, ec
+	}
+
+	if !config.ContinueOnScriptError && ec != 0 {
+		logger.Infof("script at: %s failed with error code %d", path, ec)
+		return pb.AgentErrorCode_UNHANDLED_SCRIPT_ERROR, ec
+	}
+
+	return pb.AgentErrorCode_NO_ERROR, ec
+}
+
+func listenForSnapshotRequests(address string, requestChan chan *pb.GuestMessage) {
+	for {
+		// Start hanging connection on server that feeds to channel
+		logger.Infof("attempting to connect to snapshot service at %s", address)
+		conn, err := grpc.Dial(address, grpc.WithInsecure())
+		if err != nil {
+			logger.Errorf("failed to connect: %v", err)
+			continue
+		}
+
+		c := pb.NewSnapshotServiceClient(conn)
+		ctx, cancel := context.WithCancel(context.Background())
+		guestReady := pb.GuestReady{
+			RequestServerInfo: false,
+		}
+		r, err := c.CreateConnection(ctx, &guestReady)
+		if err != nil {
+			logger.Errorf("error creating connection: %v", err)
+			continue
+		}
+		logger.Infof("created hanging connection with snapshot service")
+		for {
+			request, err := r.Recv()
+			if err != nil {
+				logger.Errorf("error reading request: %v", err)
+				break
+			}
+			logger.Infof("received snapshot request")
+			requestChan <- request
+		}
+		cancel()
+	}
+}
+
+func getSnapshotResponse(guestMessage *pb.GuestMessage) *pb.SnapshotResponse {
+	switch request := guestMessage.Msg.(type) {
+	case *pb.GuestMessage_SnapshotRequest:
+		response := &pb.SnapshotResponse{
+			OperationId: request.SnapshotRequest.GetOperationId(),
+			Type:        request.SnapshotRequest.GetType(),
+		}
+
+		config, err := getSnapshotConfig()
+		if err != nil {
+			response.AgentReturnCode = pb.AgentErrorCode_INVALID_CONFIG
+			return response
+
+		}
+
+		var url string
+		switch request.SnapshotRequest.GetType() {
+		case pb.OperationType_PRE_SNAPSHOT:
+			logger.Infof("handling pre snapshot request for operation id %d", request.SnapshotRequest.GetOperationId())
+			if seenPreSnapshotoperationIds.Contains(request.SnapshotRequest.GetOperationId()) {
+				logger.Infof("duplicate pre snapshot request operation id %d", request.SnapshotRequest.GetOperationId())
```

Comment:
does this need to be logged..?

Reply:
Think this might be helpful for our debugging if that's alright.

Reply:
then shouldn't it be logger.Debugf with Debug set to true in your testing builds? what i meant by this question is does it need to be logged on customer instances